src/typescript/frontend/src/hooks/use-node-dimensions.ts

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect, type RefObject } from "react";
+
+function getNodeDimensions(ref: RefObject<HTMLElement>) {
+  if (ref.current) {
+    const { width, height } = ref.current.getBoundingClientRect();
+    return {
+      width,
+      height,
+    };
+  }
+  return {
+    width: 0,
+    height: 0,
+  };
+}
+
+export default function useNodeDimensions(ref: RefObject<HTMLElement>) {
+  const [windowDimensions, setWindowDimensions] = useState(getNodeDimensions(ref));
+
+  useEffect(() => {
+    function handleResize() {
+      setWindowDimensions(getNodeDimensions(ref));
+    }
+
+    const element = ref?.current;
+
+    if (!element) return;
+
+    const observer = new ResizeObserver(handleResize);
+
+    observer.observe(element);
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [ref]);
+
+  return windowDimensions;
+}
```

Comment:
```suggestion
```

```react
  const [offsetHeight, setOffsetHeight] = useState(0);

  const elementRef = useCallback((node: HTMLElement | null) => {
    if (node) {
      setOffsetHeight(node.offsetHeight);
    }
  }, []);
  
  return (
    <div ref={elementRef}>
      {offsetHeight}
    </div>
  );
```


You can actually use a neat-o little trick here, where you pass a `useCallback` function to the `ref` prop. You get a stable reference to the element that also re-renders when the element ref changes (meaning your ref is always current)

I used one in the header component at one point, you can see here:

https://github.com/econia-labs/emojicoin-dot-fun/blob/210f9e8082b02fb55090ed9986443e7b6e2563c6/src/typescript/frontend/src/components/header/index.tsx#L33

Reply:
Discussed on call, we'll fix this in another PR! Too difficult to get it working right now since it requires reactively updating the element based on the window size, not just getting the element width